### PR TITLE
Enable VideoPress block on Android

### PR DESCRIPTION
**Pull request by SiobhyB**
With this PR, the VideoPress block has been registered behind a DEV flag. The block itself is not complete or fully functional, but is intended to serve as a useful placeholder to be iterated on.

## Related PRs

* `Gutenberg`: https://github.com/WordPress/gutenberg/pull/47836
* `Gutenberg Mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/5459
* `Jetpack`: https://github.com/Automattic/jetpack/pull/28812
* `iOS`: https://github.com/wordpress-mobile/WordPress-iOS/pull/20100

## Testing

Please refer to [the Gutenberg Mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5459) as the "central PR" with the most up-to-date testing instructions.

<hr />

## Regression Notes

1. Potential unintended areas of impact
2. What I did to test those areas of impact (or what existing automated tests I relied on)

As this is a new block that's been registered behind a DEV flag, it's hoped it won't have any impact on other blocks or areas of the editor. That said, it's always best to be careful and I manually smoke tested some other blocks (including the core video block) to check for any obvious issues.

3. What automated tests I added (or what prevented me from doing so)

No automated tests for now as this is the first step in a larger project. Tests related to this new block will come as it's worked on and functionality is added.

<hr />

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java [hunk 1]
```diff
@@ -348,6 +348,10 @@ public static boolean supportsTiledGalleryFeature(SiteModel site) {
         return site != null && site.isWPCom();
     }
 
+    public static boolean supportsVideoPressFeature(SiteModel site) {
+        return site != null && site.isWPCom();
```
Comment (SiobhyB) [inline WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java:352]: Enabled only on simple WordPress.com sites for now and added [a task to the project board](https://github.com/orgs/wordpress-mobile/projects/166/views/1?layout=board) for this to eventually be changed to match the web.

File: build.gradle [hunk 2]
```diff
@@ -20,7 +20,7 @@ ext {
     automatticRestVersion = '1.0.8'
     automatticStoriesVersion = '2.0.0'
     automatticTracksVersion = '2.2.0'
-    gutenbergMobileVersion = 'v1.88.0'
+    gutenbergMobileVersion = '5459-0b4b05d499197825893260921c662ed3d45a6f33'
```
Comment (fluiddot) [inline build.gradle]: We'd need to update this reference to the latest commit: `364d743d55c79c0eb462a1df79520173bcb8aab9`.

File: build.gradle [hunk 3]
```diff
@@ -20,7 +20,7 @@ ext {
     automatticRestVersion = '1.0.8'
     automatticStoriesVersion = '2.0.0'
     automatticTracksVersion = '2.2.0'
-    gutenbergMobileVersion = 'v1.88.0'
+    gutenbergMobileVersion = '5459-1e82ee9edc7a4fa5058fadfdb8da6735b94cf933'
```
Comment (fluiddot) [inline build.gradle]: @SiobhyB This reference is no longer valid as the commits in the GB-mobile PR have been recreated due to a rebase ([reference](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5459#event-8502143957)). The last commit SHA that could be used here is: `81826df1f59b0002c77e7bc6a0dc4356fed2b273`.

**Review by fluiddot — CHANGES_REQUESTED** [review]

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

I checked that the VideoPress block is displayed as unsupported and that can be edited via UBE (Unsupported Block Editor) on Atomic sites.

**NOTE:** On Simple sites when using UBE, it shows a warning due to the block not being available on the site. This is expected as the block on the web hasn't been released yet to users.

_Tested build `pr17928-0ce5520` on a Samsung Galaxy S20 FE 5G (Android 13)._

**Pull request merged**